C7GameData/Tile.cs

```diff
@@ -50,6 +50,22 @@ public MapUnit findTopDefender()
         
         public static Tile NONE = new Tile();
 
+        public bool neighborsCoast() {
+            TerrainType northeastType = neighbors[TileDirection.NORTHEAST].baseTerrainType;
+		    TerrainType northwestType = neighbors[TileDirection.NORTHWEST].baseTerrainType;
+		    TerrainType southeastType = neighbors[TileDirection.SOUTHEAST].baseTerrainType;
+    	    TerrainType southwestType = neighbors[TileDirection.SOUTHWEST].baseTerrainType;
+
+		    TerrainType[] neighborTerrains = { northeastType, northwestType, southeastType, southwestType };
```

Comment:
Perhaps having a way to easily pull out just the diagonal members of neighbors as some sort of getter would make this cleaner. I'm not exactly sure what the best approach is to that, but having to manually get each of the 4 types and then define an array to iterate through them is awkward.

Reply:
Added one.  That'll be nice to have if it turns out there is another diagonal case, which there probably will be.  It reads a bit nicer in the current _neighborsCoast_ method as well.